android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiTableViewRowProxyItem.java

```diff
@@ -649,6 +645,32 @@ public Drawable getSelectorDrawable()
 		return selectorDrawable;
 	}
 
+	public void populateRectDict(KrollDict d)
+	{
+		if (content != null) {
+			int position[] = new int[2];
+			content.getLocationInWindow(position);
+
+			TiDimension nativeWidth = new TiDimension(content.getWidth(), TiDimension.TYPE_WIDTH);
```

Comment:
probably want to explicitly state that these values are in pixels by calling `setUnits(TypedValue.COMPLEX_UNIT_PX)`. I think this happens to end up being treated as pixels solely by some luck internally.

On that note, it is absurd we do not have a constructor that takes in the unit types on `TiDimension`. We should add one.

Reply:
@drauggres I've added the TiDimension constructor that accepts the units as the third argument here (on master branch): 
https://github.com/appcelerator/titanium_mobile/commit/84fc4f2b3db74531d40cf959864a9a4850cef943

Reply:
> I think this happens to end up being treated as pixels solely by some luck internally.

On Android, the default native coordinates are pixels. This is the assumed units when accessing Google's Java APIs. Our `TiDimension` class was designed to default to pixels as well. So, it's not luck. Although it does this in a rather obtuse manner in this case, that I agree with. I would prefer the units to be explicit set as well.

Honestly, I don't like the design of our Java `TiDimension` class anyways. It's fine for storing coordinates received from JavaScript. But for unit coordinate conversion like this, it's cumbersome to use since it wants to be fed a left/top/right/bottom type which is only relevant to percentage based units. In the future, we should really extract the conversion code to a separate class for convenience.

Reply:
This method is "95% copypaste" of [TiViewProxy.getRect()](https://github.com/appcelerator/titanium_mobile/blob/master/android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java#L271)

Reply:
For some reason (I don't know it) TableViewRowProxy does not save its "native view" to `view` property and uses additional property `content`, so `super` method can't calculate "rect" because `view` is `null`.

Reply:
The Android team already knows that the default units are pixels. I can go either or on this.

Passing the decor view to `TiDimension` is the right move here. It only needs it to fetch the screen DPI in this case. The proxy's actual view is only needed for percentage units, which is never needed here (that was my gripe with the `TiDimension` class up above). The decor view can be `null` if the activity was destroyed. So, the `null` case can theoretically happen when backing out of the app and you fetch the row's rect via the activity's `onDestroy` listener. Although with the changes we've made in Titanium 8.0.0 already, this may no longer be an issue anymore. But it's better to be safe than sorry. I'm fine with the `null` check.

Reply:
I moved part of  `TiViewProxy.getRect()` to `TiViewProxy.getViewRect(View v)` and changed `TableViewRowProxy.getRect()` to use it.

I didn't understand everything above about pixels, dpi, etc, but now all suspicious code again in [one place](https://github.com/appcelerator/titanium_mobile/pull/10542/files#diff-bec22e2efdd8b352a7754ea42f435dc7R281), and you can deal with it regardless of this PR.

Reply:
This discussion is not related anymore for this PR, so I'm closing it as resolved.